Export composite-get-overlay-window, and document it

equwal opened the pull request:
Exported this (since it is needed anyway for the other exported symbol `composite-release-overlay-window`). 

Added some documentation strings to these.

JMC-design commented:
Looks fine to me.  

There should be a warning added somewhere. If no composite manager is running then the client requesting the overlay window becomes responsible for redrawing everything.  If the client cannot setup a pass through input on the overlay window, which they currently cannot because I haven't pushed xfixes yet, then the overlay will prevent any clients from receiving input.
So without a composite manager already running the overlay window is not usable.

equwal commented:
> Looks fine to me.
> 
> There should be a warning added somewhere. If no composite manager is running then the client requesting the overlay window becomes responsible for redrawing everything. If the client cannot setup a pass through input on the overlay window, which they currently cannot because I haven't pushed xfixes yet, then the overlay will prevent any clients from receiving input.
> So without a composite manager already running the overlay window is not usable.

I hadn't considered that use case, since I am working on the window manager (not applications usage).

I added a relevant note in this docstring, and added a few more docstrings.

dkochmanski commented on extensions/composite.lisp:
```diff
@@ -25,28 +25,34 @@
 	  composite-get-overlay-window
-	  
-	  ))
+    composite-release-overlay-window))
```
please detabify and reindent the file in a separate commit

dkochmanski commented on extensions/composite.lisp:
```diff
@@ -25,28 +25,34 @@
+(defconstant +redirect-automatic+ 0
+  "The automatic update type automatically updates the parent window.")
+(defconstant +redirect-manual+ 1
+  "Prevents some activities that would otherwise be automatic.")
```
please make all defconstants consistent. it is up to you, either:
- arrange them in columns like they were (but not with tabs)
- keep them in a single line but without arranging them in columns
- break the line after the value (so docstring is in a separate line)

if you don't care then go after the last option.

dkochmanski commented:
lgtm, thanks!
